### ISSUE TITLE
WAGED NPE occurs when new resource is added while the cluster requires rebalance overwrite

### ISSUE THREAD

**Issue by GrantPSpencer** [apache/helix#2909]
### Describe the bug
**TLDR:**
NPE occurs when new WAGED delay rebalance enabled resource is added to a cluster that requires delay rebalance overwrites. The cluster cannot recover and pipeline runs will fail until delay rebalance overwrites are no longer needed (typically that down nodes are now outside the delay window)

**Description:**
When a new resource is added and `_partialRebalanceRunner.isAsyncPartialRebalanceEnabled()` evaluates to true, `handleDelayedRebalanceMinActiveReplica` is called with a `resourceMap` that contains the new resource while the `currentResourceAssignment` does not. 

This causes an NPE down the line when `generateClusterModelForDelayedRebalanceOverwrites` is called. This method is called only when there are instances down within the delayed rebalance window, more specifically when this `activeNodes.equals(enabledLiveInstances) || !requireRebalanceOverwrite(clusterData, currentResourceAssignment)` evaluates to false. 

The NPE occurs in `findToBeAssignedReplicasForMinActiveReplica` because we iterate over the keys in the resourceMap (which contains the new resource) and use that to access the resourceAssignments in the `currentAssignment` (previously called the `currentResourceAssignment`). We then get a null resourceAssignment and try to operate on it 
```
resourceAssignment.getMappedPartitions().forEach(partition ->
    resourceAssignment.getReplicaMap(partition).forEach((logicalId, state) ->
          ...
```

This is the NPE stack trace:
```
17157915:Caused by: java.lang.NullPointerException: Cannot invoke "org.apache.helix.model.ResourceAssignment.getMappedPartitions()" because "resourceAssignment" is null
17157916-	at org.apache.helix.controller.rebalancer.util.DelayedRebalanceUtil.findToBeAssignedReplicasForMinActiveReplica(DelayedRebalanceUtil.java:332) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157917-	at org.apache.helix.controller.rebalancer.waged.model.ClusterModelProvider.generateClusterModel(ClusterModelProvider.java:257) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157918-	at org.apache.helix.controller.rebalancer.waged.model.ClusterModelProvider.generateClusterModelForDelayedRebalanceOverwrites(ClusterModelProvider.java:82) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157919-	at org.apache.helix.controller.rebalancer.waged.WagedRebalancer.handleDelayedRebalanceMinActiveReplica(WagedRebalancer.java:415) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157920-	... 10 more
```

### To Reproduce
1. Set up cluster with delay rebalance enabled (make it large enough so nodes can go down within the window)
2. Add a WAGED resource (possibly dont need to do this part)
3. Disable enough instances that min_active is violated for at least one replica
4. Within the delay rebalance window, attempt to add a new WAGED resource
5. NPE! 🥳 

I was able to reproduce this locally with the test added in this branch:
https://github.com/GrantPSpencer/helix/pull/34

### Expected behavior
Controller should handle this possible null. Either it does not consider newly added resources (takes a union of the keys in `resources` and `currentAssignments`) or skips calculation if the assignment is null. 

### Additional context
Full stack trace for rebalance failure
```
17157902-2024/09/09 17:26:48.380 ERROR [WagedRebalancer] [HelixController-pipeline-default-Ambry-video-(1a7fd4c4_DEFAULT)] [helix] [] Failed to calculate the new assignments.
17157903-org.apache.helix.HelixRebalanceException: Failed to compute for delayed rebalance overwrites in cluster ZnRecord=Ambry-video, {DELAY_REBALANCE_ENABLED=true, DELAY_REBALANCE_TIME=21600000, ERROR_OR_RECOVERY_PARTITION_THRESHOLD_FOR_LOAD_BALANCE=2000, FAULT_ZONE_TYPE=rack, PERSIST_BEST_POSSIBLE_ASSIGNMENT=true, STATE_TRANSITION.maxThreads=300, TOPOLOGY=/rack/host, TOPOLOGY_AWARE_ENABLED=true}{DEFAULT_PARTITION_WEIGHT_MAP={DISK=386}, REBALANCE_PREFERENCE={EVENNESS=1, LESS_MOVEMENT=4}}{INSTANCE_CAPACITY_KEYS=[DISK], STATE_TRANSITION_THROTTLE_CONFIGS=[{"THROTTLE_SCOPE":"RESOURCE","MAX_PARTITION_IN_TRANSITION":"1500","REBALANCE_TYPE":"LOAD_BALANCE"}]}, Stat=Stat {_version=5, _creationTime=1592007259555, _modifiedTime=1715720590967, _ephemeralOwner=0} Failure Type: INVALID_CLUSTER_STATUS
17157904-	at org.apache.helix.controller.rebalancer.waged.WagedRebalancer.handleDelayedRebalanceMinActiveReplica(WagedRebalancer.java:428) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157905-	at org.apache.helix.controller.rebalancer.waged.WagedRebalancer.emergencyRebalance(WagedRebalancer.java:501) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157906-	at org.apache.helix.controller.rebalancer.waged.WagedRebalancer.computeBestPossibleAssignment(WagedRebalancer.java:339) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157907-	at org.apache.helix.controller.rebalancer.waged.WagedRebalancer.computeBestPossibleStates(WagedRebalancer.java:316) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157908-	at org.apache.helix.controller.rebalancer.waged.WagedRebalancer.computeNewIdealStates(WagedRebalancer.java:248) [org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157909-	at org.apache.helix.controller.stages.BestPossibleStateCalcStage.computeResourceBestPossibleStateWithWagedRebalancer(BestPossibleStateCalcStage.java:445) [org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157910-	at org.apache.helix.controller.stages.BestPossibleStateCalcStage.compute(BestPossibleStateCalcStage.java:289) [org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157911-	at org.apache.helix.controller.stages.BestPossibleStateCalcStage.process(BestPossibleStateCalcStage.java:94) [org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157912-	at org.apache.helix.controller.pipeline.Pipeline.handle(Pipeline.java:75) [org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157913-	at org.apache.helix.controller.GenericHelixController.handleEvent(GenericHelixController.java:903) [org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157914-	at org.apache.helix.controller.GenericHelixController$ClusterEventProcessor.run(GenericHelixController.java:1554) [org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157915:Caused by: java.lang.NullPointerException: Cannot invoke "org.apache.helix.model.ResourceAssignment.getMappedPartitions()" because "resourceAssignment" is null
17157916-	at org.apache.helix.controller.rebalancer.util.DelayedRebalanceUtil.findToBeAssignedReplicasForMinActiveReplica(DelayedRebalanceUtil.java:332) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157917-	at org.apache.helix.controller.rebalancer.waged.model.ClusterModelProvider.generateClusterModel(ClusterModelProvider.java:257) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157918-	at org.apache.helix.controller.rebalancer.waged.model.ClusterModelProvider.generateClusterModelForDelayedRebalanceOverwrites(ClusterModelProvider.java:82) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157919-	at org.apache.helix.controller.rebalancer.waged.WagedRebalancer.handleDelayedRebalanceMinActiveReplica(WagedRebalancer.java:415) ~[org.apache.helix.helix-core-1.4.1-dev-202408061926.jar:1.4.1-dev-202408061926]
17157920-	... 10 more
```
